system_config/src/SystemConfig.cpp

```diff
@@ -14,22 +14,14 @@ using std::shared_ptr;
 using std::string;
 using std::vector;
 
-// Initialize static variables
-std::string SystemConfig::rootPath;
-std::string SystemConfig::logPath;
-std::string SystemConfig::configPath;
-std::string SystemConfig::hostname;
-std::mutex SystemConfig::configsMapMutex;
-std::map<std::string, std::shared_ptr<Configuration>> SystemConfig::configs;
-
 /**
  * The method for getting the singleton instance.
  * @return A pointer to the SystemConfig object, you must not delete.
  */
-SystemConfig* SystemConfig::getInstance()
+SystemConfig& SystemConfig::getInstance()
 {
     static SystemConfig instance;
```

Comment:
I'm really not a friend of this singleton-as-static-local pattern, but oh well

Reply:
Can make it a static pointer and delete on shutdown call. Problem is that not everyone is calling shutdown.